Sources/ApolloCodegenLib/Frontend/JavaScript/src/validationRules.ts

```diff
@@ -65,6 +67,25 @@ export function NoTypenameAlias(context: ValidationContext) {
   };
 }
 
+export function DeferredInlineFragmentNoTypeCondition(context: ValidationContext) {
+  return {
+    InlineFragment(node: InlineFragmentNode) {
+      if (node.directives) {
+        for (const directive of node.directives) {
+          if (directive.name.value == "defer" && node.typeCondition == undefined) {
```

Comment:
`"defer"` should already be available in `GraphQLDeferDirective.name`. Probably be better to use that than duplicate strings.